### PR TITLE
Trigger rebuild when schema changes.

### LINKED ISSUE #1175: Trigger re-build when schema changes.
There is a common pitfall when working on an indexer: modifying only the schema does not trigger a rebuild of the WASM binary.

Example:

1. deploy an indexer:
```
cargo run -p forc-index -- deploy --path examples/hello-world/hello-indexer/ --target-dir `pwd` --replace-indexer --remove-data
```

2. modify the schema:
```
echo "type Ty { id: ID! }" >> examples/hello-world/hello-indexer/schema/hello_indexer.schema.graphql
```

3. redeploy:
```
cargo run -p forc-index -- deploy --path examples/hello-world/hello-indexer/ --target-dir `pwd` --replace-indexer --remove-data
```

This triggers an error loop on `fuel-indexer`:

```
2023-08-02T09:21:59.865028Z ERROR fuel_indexer::database: 143: TypeId(-8293274664658733968) not found in tables: {}.

Does the schema version in SchemaManager::new_schema match the schema version in Database::load_schema?

Do your WASM modules need to be rebuilt?
```

The common advice is to modify `lib.rs` after changing the schema. This is a bad developer experience.

We should find a way to trigger rebuild when only the schema changes.

#### Option 1: `include_str!`: include the schema file contents in the WASM binary

`include_str!(file)` adds the `file` as a dependency. Thus, if we include this in the output of the `indexer` macro: 

```
    let mut output = quote! {
        #namespace_tokens
        #identifer_tokens
        #version_tokens
        static SCHEMA: &str = include_str!(#path_str);
    };
```

then the indexer will be rebuilt when the schema changes.

#### Option 2: always `touch src/lib.rs`

We always (unless `--skip-build` is set) modify the `atime` and `mtime` of `src/lib.rs`. This will always trigger a rebuild, even if the schema, ' lib.rs`, or any dependency has not changed.

I think removing the schema pitfall is worth it.

### PR DESCRIPTION
**Pull request by lostman**
### Description

Closes #1175.

When the schema is modified, rebuild the WASM module.

### Testing steps

Create and deploy an indexer:

```
forc index new --namespace test test-indexer
cd test-indexer
forc index deploy
```

Change the schema and deploy again:

```
echo "type Type { id: ID! }" >> schema/test_indexer.schema.graphql
forc index deploy
```

Without this PR, the second `forc index deploy` would not rebuild the WASM module, and the deployed indexer would cause an error.

### Changelog

* add ensure_rebuild_if_schema_changed

### CODE REVIEW

**Comment by ra0x3** [comment]
@lostman 
- Commit title does not conform to the CONTRIBUTING guidelines
- Also your testing steps mention `forc index` which is the published plugin, when that should be `forc-index`
- Testing steps should also mention that you have to rebuild the plugin locally to test

**Review by ra0x3 — APPROVED** [review]
Works, thanks @lostman this has been a PITA for a long time. Glad to see it solved.

**Comment by ra0x3** [comment]
Let's defer to @deekerno before merging

File: plugins/forc-index/src/ops/forc_index_build.rs
Comment (deekerno) [inline plugins/forc-index/src/ops/forc_index_build.rs]: Can we use `std::env::current_dir()` instead of `".".into()` as it's platform-agnostic?

File: plugins/forc-index/src/ops/forc_index_deploy.rs
Comment (deekerno) [inline plugins/forc-index/src/ops/forc_index_deploy.rs]: s/`manifest_modulue_file`/`manifest_module_file`/g

**Review by deekerno — COMMENTED** [review]
Looks pretty good to me! Left just two nits.

**Comment by lostman** [comment]
@deekerno fixed the nits.

**Review by deekerno — APPROVED** [review]

**Pull request merged**
